src/plugins/controls/public/control_group/editor/control_editor.tsx

```diff
@@ -163,51 +200,107 @@ export const ControlEditor = ({
       </EuiFlyoutHeader>
       <EuiFlyoutBody data-test-subj="control-editor-flyout">
         <EuiForm>
+          <EuiFormRow label={ControlGroupStrings.manageControl.getDataViewTitle()}>
+            <DataViewPicker
+              dataViews={state.dataViewListItems}
+              selectedDataViewId={dataView?.id}
+              onChangeDataViewId={(dataViewId) => {
+                setLastUsedDataViewId?.(dataViewId);
+                if (dataViewId === dataView?.id) return;
+
+                onTypeEditorChange({ dataViewId });
+                setSelectedField(undefined);
+                get(dataViewId).then((newDataView) => {
+                  getCompatibleControlTypes(newDataView);
+                  setState((s) => ({ ...s, selectedDataView: newDataView }));
+                });
+              }}
+              trigger={{
+                label:
+                  state.selectedDataView?.title ??
+                  ControlGroupStrings.manageControl.getSelectDataViewMessage(),
+              }}
+            />
+          </EuiFormRow>
+          <EuiFormRow label={ControlGroupStrings.manageControl.getFieldTitle()}>
+            <FieldPicker
+              filterPredicate={(field: DataViewField) => {
+                return state.fieldRegistry ? Boolean(state.fieldRegistry[field.name]) : false;
```

Comment:
A ternary might be overkill here, a more readable way to put this could be

 `return Boolean(state?.fieldRegistry?.[field.name]);`